src/eos-updater-branch-file.c

```diff
@@ -0,0 +1,309 @@
+/* -*- mode: C; c-file-style: "gnu"; indent-tabs-mode: nil; -*-
+ *
+ * Copyright 2016 Kinvolk GmbH
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the
+ * Free Software Foundation, Inc., 59 Temple Place - Suite 330,
+ * Boston, MA 02111-1307, USA.
+ *
+ * Author: Krzesimir Nowak <krzesimir@kinvolk.io>
+ */
+
+#include "eos-updater-branch-file.h"
+
+#include "eos-util.h"
+
+static void
+eos_branch_file_dispose_impl (EosBranchFile *bf)
+{
+  g_clear_pointer (&bf->raw_contents, g_bytes_unref);
+  g_clear_pointer (&bf->raw_signature, g_bytes_unref);
+  g_clear_pointer (&bf->branch_file, g_key_file_unref);
+  g_clear_pointer (&bf->download_time, g_date_time_unref);
+}
+
+static void
+eos_branch_file_finalize_impl (EosBranchFile *bf)
+{
+  g_free (bf->contents_sha512sum);
+}
+
+EOS_DEFINE_REFCOUNTED (EOS_BRANCH_FILE,
+                       EosBranchFile,
+                       eos_branch_file,
+                       eos_branch_file_dispose_impl,
+                       eos_branch_file_finalize_impl)
+
+static GDateTime *
+get_download_time (GFile *file,
+                   GCancellable *cancellable,
+                   GError **error)
+{
+  g_autoptr(GFileInfo) info = NULL;
+  guint64 mod_time;
+
+  info = g_file_query_info (file,
+                            G_FILE_ATTRIBUTE_TIME_MODIFIED,
+                            G_FILE_QUERY_INFO_NONE,
+                            cancellable,
+                            error);
+  if (info == NULL)
+    return NULL;
+
+  mod_time = g_file_info_get_attribute_uint64 (info,
+                                               G_FILE_ATTRIBUTE_TIME_MODIFIED);
+  return g_date_time_new_from_unix_utc (mod_time);
+}
+
+static gboolean
+read_file_to_bytes (GFile *file,
+                    GCancellable *cancellable,
+                    GBytes **out_bytes,
+                    GError **error)
+{
+  g_autofree gchar *contents = NULL;
+  gsize len = 0;
+
+  if (!g_file_query_exists (file, cancellable))
+    return TRUE;
```

Comment:
Is there any reason why this returns true if the file does not exist? As far as I can tell, this function is later used by `eos_branch_file_new_from_files` which just errors out if the file failed to be read. It might be better to just remove this and attempt to read the file, then use whatever error GIO throws up if that fails.